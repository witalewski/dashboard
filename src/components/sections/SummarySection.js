import React from 'react';
import styled from '@emotion/styled';
import { format } from 'date-fns';
import { Section } from '../Section';
import { HorizontalBar } from '../charts/HorizontalBar';

const SummaryContentStyled = styled.div`
  min-width: 200px;

  display: flex;
  flex-direction: column;
  justify-content: center;
`;

export const SummarySection = ({ title, className, data }) => (
  <Section title={title} className={className}>
    <SummaryContentStyled>
      <div className="current-month-summary__label">
        In {format(new Date(), 'MMMM')} you did
      </div>
      <div className="current-month-summary__sum">${data.revenueThisMonth}</div>
      <div className="comparison">
        <div className="comparison__label">Last month you did:</div>
        <div className="comparison__sum">${data.revenueLastMonth}</div>
      </div>
      <div className="comparison">
        <div className="comparison__label">Average monthly revenue:</div>
        <div className="comparison__sum">${data.revenueAverage}</div>
      </div>
      <HorizontalBar percent={data.revenueRate * 100} />
      <div className="percent-of-average-label">
        of your monthly average revenue
      </div>
    </SummaryContentStyled>
  </Section>
);
