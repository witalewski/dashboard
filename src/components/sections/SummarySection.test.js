import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import mockChartData from '../../fixtures/mockChartData.json';
import { SummarySection } from './SummarySection';

describe('SummarySection', () => {
  it('matches snapshot with mock data', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
        <SummarySection
        className="summary-section-wrapper"
        title="Summary"
        data={mockChartData.summaryChart.data}
      />
    );
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});
