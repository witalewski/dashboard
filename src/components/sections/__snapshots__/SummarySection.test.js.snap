// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SummarySection matches snapshot with mock data 1`] = `
<Section
  className="summary-section-wrapper"
  title="Summary"
>
  <ForwardRef(render)>
    <div
      className="current-month-summary__label"
    >
      In 
      December
       you did
    </div>
    <div
      className="current-month-summary__sum"
    >
      $
      100.64
    </div>
    <div
      className="comparison"
    >
      <div
        className="comparison__label"
      >
        Last month you did:
      </div>
      <div
        className="comparison__sum"
      >
        $
        199.09
      </div>
    </div>
    <div
      className="comparison"
    >
      <div
        className="comparison__label"
      >
        Average monthly revenue:
      </div>
      <div
        className="comparison__sum"
      >
        $
        131.11
      </div>
    </div>
    <HorizontalBar
      percent={38}
    />
    <div
      className="percent-of-average-label"
    >
      of your monthly average revenue
    </div>
  </ForwardRef(render)>
</Section>
`;
